src/main/kotlin/org/codefreak/codefreak/service/file/FileCollectionRepositoryService.kt

```
@@ -0,0 +1,238 @@
+    val status = git.status().call()
+
+    val missing = status.missing
+    if (missing.isNotEmpty()) {
+      return true
+    }
+
+    val modified = status.modified
+    if (modified.isNotEmpty()) {
+      return true
+    }
+
+    val untracked = status.untracked
+    if (untracked.isNotEmpty()) {
+      return true
+    }
+
+    val untrackedFolders = status.untrackedFolders
+    if (untrackedFolders.isNotEmpty()) {
+      return true
+    }
+    return false
+  }
+
+  /**
+   * Saves the current used version (git sha) to the answer table in the database.
+   * @param git the Jgit git instance to operate on
+   * @param collectionID is the Id of the collection. The id is the same as the answer id
+   */
+  private fun setCurrentHeadCommitId(git: Git, collectionID: UUID) {
+    val latestCommit = Git(git.repository).log().setMaxCount(1).call().iterator().next()
+    answerService.saveCommitId(collectionID, latestCommit.name)
+  }
+
+  override fun resetAndLoadVersion(collectionID: UUID, changeToCommitId: String, commitMessage: String): Boolean {
+    val workingTreePath = createCollectionPath(collectionID)
+    try {
+      val git = getGit(collectionID)
+      val commitId = answerService.getCommitId(collectionID)
+      if (commitId != changeToCommitId) {
+        if (gotTouchedForReset(git)) {
+          saveChanges(collectionID, "$commitMessage autosave", true)
+        }
+        FileUtils.cleanDirectory(workingTreePath.toFile())
+        git.checkout().setStartPoint(changeToCommitId).setAllPaths(true).call()
+        val missingFiles = git.status().call().missing
+        for (remove in missingFiles) {
+          git.rm().addFilepattern(remove).call()
+        }
+        git.add().addFilepattern(".").call()
+        answerService.saveCommitId(collectionID, changeToCommitId)
+        return true
+      }
+      return false
+    } catch (err: CanceledException) {
+      return false
```

question: what does this exception mean? should we swallow it?